TermPrinter: Use Referent for terms

hojberg opened the pull request:
## Overview
Fix a bug which manifested itself in the SyntaxText of the Codebase
server where references to terms, where using TypeReference instead of
TermReference. 

Fixes: https://github.com/unisonweb/unison/issues/2154

## Implementation notes
This was due to using a Reference instead of a Referent.

## Loose ends
@pchiusano you'd mentioned on https://github.com/unisonweb/unison/issues/2154 to change the constructors to be more explicit, but I wasn't sure exactly which constructors might change.

hojberg committed:
TermPrinter: Use Referent for terms

Fix a bug which manifested itself in the SyntaxText of the Codebase
server where references to terms, where using TypeReference instead of
TermReference. This was due to using a Reference instead of a Referent.

pchiusano reviewed (approved):
@hojberg Cool. 

I was just saying that I think the name of that `SyntaxText` constructor is confusing. It's just called `Reference` when it really means "type reference". I'd just rename it to `TypeReference`. The confusing name is probably what led to this problem in the first place. :) 

Fixing would entail editing `Unison.Util.SyntaxText` to rename that constructor, and then fix any compile errors. Should be quick but not a big deal if you'd rather not futz with it now or think it belongs as a separate PR.

hojberg committed:
SyntaxText: Rename Referent/Reference

Rename the constructors of `SyntaxText.Element`:

  `Referent`  -> `TermReference`
  `Reference` -> `TypeReference`

hojberg commented:
@pchiusano updated with new constructor names 🎉
